Modules/HypnokinkModule/Scenes/Encounters/HK_HypnoEncounterFetch.gd

```diff
@@ -0,0 +1,296 @@
+		npcVariation = "subby"
+		
+	startLocation = GM.pc.getLocation()
+	
+	var eligibleEndLocations = []
+	for loc in possibleEndLocations:
+		var candidatePath = GM.world.calculatePath(startLocation, loc)
+		if(candidatePath.size() > 4): #avoid nonexistent or too short paths
+			var pathWeight = 1.0 / candidatePath.size()
+			eligibleEndLocations.append([loc, pathWeight])
+	endLocation = RNG.pickWeightedPairs(eligibleEndLocations)
+	path = GM.world.calculatePath(startLocation, endLocation)
+	
+func resolveCustomCharacterName(_charID):
+	if(_charID == "npc"):
+		return npcID
+
+func _run():
+	var npc: BaseCharacter = GlobalRegistry.getCharacter(npcID)
+	npc = npc #dont bother me godot		
+	
+	if(state == ""):
+		addCharacter(npcID)
+		playAnimation(StageScene.Duo, "stand", {npc=npcID})
+		# (default)
+		if(!(npcVariation in ["mean", "kind", "subby"])):
+
+			saynn("[say=npc]"+RNG.pick([
+				"You like being useful, right?",
+				"I've got something for you to do.",
+				"You want to serve me?",
+			])+"[/say]")
+			
+		# (mean version)
+		if(npcVariation == "mean"):
+
+			saynn("[say=npc]"+RNG.pick([
+				"You're gonna do something for me.",
+				"Make yourself useful.",
+				"Time to serve, slut."
+			])+"[/say]")
+
+		# (kind version)
+		if(npcVariation == "kind"):
+			
+			saynn("[say=npc]"+RNG.pick([
+				"You have some time, right?",
+				"You wouldn't mind doing something for me, right?",
+			])+"[/say]")
+			
+		# (subby version)
+		if(npcVariation == "subby"):
+			saynn("[say=npc]"+RNG.pick([
+				"Please, can you help me?",
+				"Can you help me out~?",
+				"Would you do something for me? I'll make it worth your while...",
+			])+"[/say]")
+			
+		saynn("Your thoughs murk over with the desire to serve.")
+			
+		saynn("[say=pc]Yes...[/say]")
+		
+		if(npcVariation in ["kind", "subby"]):
+			saynn("[say=npc]"+descriptionsPolite[endLocation]+"[/say]")
+		elif(npcVariation == "mean"):
+			saynn("[say=npc]"+descriptionsDemanding[endLocation]+"[/say]")
+		else:
+			if(RNG.randi_range(0,100) > 50):
+				saynn("[say=npc]"+descriptionsPolite[endLocation]+"[/say]")
+			else:
+				saynn("[say=npc]"+descriptionsDemanding[endLocation]+"[/say]")
+		
+		saynn("Compelled to obey, you...")
+		HypnokinkUtil.addUnresistableHypnoButtons(self, "moving_towards", ["Head out", "Get moving", "Fetch"])
+	
+	if(state == "moving_towards" || state == "moving_back"):
+		removeCharacter(npcID)
+		playAnimation(StageScene.Solo, "walk")
+		if(path.size() > 0):
+			aimCameraAndSetLocName(path[0])
+		
+		var _roomInfo = GM.world.getRoomByID(path[0])
+		
+		var adjective = RNG.pick(["purposefully", "obediently", "docilely", "slavishly"])
+		saynn("You "+adjective+" walk towards your destination.")
+		
+		if(_roomInfo == null):
+			pass
+		else:
+			if(GM.pc.isBlindfolded() && !GM.pc.canHandleBlindness()):
+				saynn(_roomInfo.getBlindDescription())
+			else:
+				saynn(_roomInfo.getDescription())
+				
+		addButton("Skip", "Fade into trance", "skip_towards")
```

Comment:
It doesn't skip if you're returning back.

Reply:
Fix'd.